src/Restore.php

```diff
@@ -700,18 +699,36 @@ private function restoreTypedTable(array $tableInfo): string
                 }
                 $columnMetadata[$metadata['key']] = $metadata['value'];
             }
-
-            $definition = [
-                'type' => $columnMetadata['KBC.datatype.type'],
-                'nullable' => $columnMetadata['KBC.datatype.nullable'] === '1',
-            ];
-            if (isset($columnMetadata['KBC.datatype.length'])) {
-                $definition['length'] = $columnMetadata['KBC.datatype.length'];
-            }
-            if (isset($columnMetadata['KBC.datatype.default'])) {
-                $definition['default'] = $columnMetadata['KBC.datatype.default'];
+            if ($destinationBucketBackendType !== $tableInfo['bucket']['backend']) {
+                $sourceBaseType = $this->getBaseType(
+                    $tableInfo['bucket']['backend'],
+                    $columnMetadata['KBC.datatype.type'],
+                );
+                switch ($destinationBucketBackendType) {
+                    case 'snowflake':
+                        $definition = (Snowflake::getDefinitionForBasetype((string) $sourceBaseType))->toArray();
+                        break;
+                    case 'bigquery':
+                        $definition = (Bigquery::getDefinitionForBasetype((string) $sourceBaseType))->toArray();
+                        break;
+                    default:
+                        $this->logger->warning('unsupported typed backend type');
+                        continue 2;
+                }
+                // the nullable property is required for PK compatability
+                $definition['nullable'] = $columnMetadata['KBC.datatype.nullable'] === '1';
```

Comment:
length a default nebudeme tady nastavovat?

Reply:
jsem tam daval length ale spadnul na timestamp co ma length v snowflake ale ne v bq

Reply:
hmm, ale u těch ostatních je škoda to zahazovat když to někdo zadá

Reply:
jo, souhlasim,  mozna muzem to zlepsit do budoustnost